### PR TITLE
feat: `<StandaloneLink />` component

### PR DESCRIPTION
**Pull request by alexcarpenter**
🎟️ [Asana Task](https://app.asana.com/0/1187888078520425/1202966328432329/f)
🔍 [Preview Link](https://react-components-git-acstandalone-link-hashicorp.vercel.app/components/standalonelink)

---

## Description

Adds new `<StandaloneLink />` component.

Figma: https://www.figma.com/file/7cYgDM618stjYUHDqAfRec/Components?node-id=2824%3A11407

### PR Checklist 🚀

Items in this checklist may not may not apply to your PR, but please consider each item carefully.

- [x] Add Asana and Preview links above.
- [x] Conduct thorough self-review.
- [x] Add or update tests as appropriate.
- [x] Conduct reasonable cross browser testing for both compatibility and responsive behavior (We have a [Sauce Labs](https://app.saucelabs.com/) account for this, if you don't have access, just ask!).
- [x] Conduct reasonable accessibility review (use the [WAS](https://accessible.org/Web-Accessibility-Standards-WAS-2.pdf) as a guide or an [axe browser plugin](https://www.deque.com/axe/) until we establish more formal checks).
- [x] Identify (in the description above) and document (add Asana tasks on [this board](https://app.asana.com/0/1100423001970639/list)) any technical debt that you're aware of, but are not addressing as part of this PR.

### CODE REVIEW

File: packages/standalone-link/style.module.css [hunk 1]
```diff
@@ -0,0 +1,117 @@
...
+    }
+  }
+}
+
+/*
+ * Arrow icon
+ */
+
+.icon {
+  overflow: visible !important;
```
Comment (alexcarpenter) [inline packages/standalone-link/style.module.css:87]: Needed to translate the arrow path.

**Comment by rayelder** [comment]
**Design feedback:**
- The chevron to arrow animation feels too quick and without easing
- `gap` between label and chevron should be 8px (My error in the Figma file, sorry)
- `color` of rollover states on `appearance: dark` should not be light blue
- Do we need `theme: tertiary`?

**Comment by alexcarpenter** [comment]
@rayelder 

> The chevron to arrow animation feels too quick and without easing
We do have easing in place https://github.com/hashicorp/react-components/pull/723/files#diff-28d3a016e297787a957fbc7b4149f008d418fdc2aa433406e21e5348d81ac764R3

> color of rollover states on appearance: dark should not be light blue
Fixed, good catch!

> Do we need theme: tertiary?
You have three themes shown in the design file?

image

**Comment by alexcarpenter** [comment]
Design approval: https://hashicorp.slack.com/archives/C02U59WGBCK/p1663190378404199?thread_ts=1663082962.871769&cid=C02U59WGBCK

File: packages/standalone-link/docs.mdx [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+## Rules
+
+- They may appear on their own, directly following content, or in combination with a primary Button
+
+## UX writing guidelines
+
+- Target a maximum of 2-3 words.
+- Target a readability grade level score of 8 to meet plain language guidelines.
+- Text uses Sentence case capitalization.
```
Comment (EnMod) [inline packages/standalone-link/docs.mdx:21]: 🙌 Oooh these are good to include for not only us, but for guiding external parties as well (which is the main benefit, I think)

**Review by EnMod — APPROVED** [review]
Digging this! Especially that chevron-to-arrow animation 🤩

**Pull request merged**